ToArcGISEarth/ToArcGISEarth/AddLayerCheckBox.cs

```diff
@@ -0,0 +1,92 @@
+﻿using ArcGIS.Desktop.Framework.Contracts;
+using ArcGIS.Desktop.Mapping;
+using ArcGIS.Desktop.Mapping.Events;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+
+namespace ToArcGISEarth
+{
+    public class AddLayerCheckBox : CheckBox
+    {
+        protected override void OnClick()
+        {
+            if ((bool)IsChecked)
+            {
+                if (!ConnectToArcGISEarthButton.IsConnectSuccessful)
+                {
+                    ArcGIS.Desktop.Framework.Dialogs.MessageBox.Show("Please connect to ArcGIS Earth");
+                    this.IsChecked = false;
+                    return;
+                }
+                else
+                {
+                    LayersAddedEvent.Subscribe(AddLayer, false);
+                }
+            }
+            else
+            {
+                LayersAddedEvent.Unsubscribe(AddLayer);
+            }
+        }
+
+        private void AddLayer(LayerEventsArgs args)
+        {
+            try
+            {
+                List<Layer> layerList = args.Layers as List<Layer>;
+                if (layerList != null && layerList.Count != 0)
+                {
+                    Layer layer = layerList[0] as Layer;
+                    string url = GetLayerUrl(layer);
+                    if (url != null)
+                    {
+                        JObject addLayerJson = new JObject
+                        {
+                            ["URI"] = url
+                        };
+                        if (layer.MapLayerType == ArcGIS.Core.CIM.MapLayerType.Operational)
+                        {
+                            addLayerJson["target"] = "OperationalLayers";
+                        }
+                        if (layer.MapLayerType == ArcGIS.Core.CIM.MapLayerType.BasemapBackground)
+                        {
+                            addLayerJson["target"] = "BasemapLayers";
+                        }
+                        string currentJson = addLayerJson.ToString();
+                        if (currentJson != null)
```

Comment:
If you don't want output string cotain formatting. You can use ToString(Formatting.None)